src/main/webapp/resources/js/pages/projects/samples/components/MergeModal.jsx

```diff
@@ -0,0 +1,182 @@
+import {
+  Alert,
+  Checkbox,
+  Col,
+  Form,
+  Input,
+  Modal,
+  notification,
+  Radio,
+  Row,
+  Space,
+} from "antd";
+import React from "react";
+import { useSelector } from "react-redux";
+import { serverValidateSampleName } from "../../../../utilities/validation-utilities";
+import { useMergeMutation } from "../../../../apis/projects/samples";
+import LockedSamplesList from "./LockedSamplesList";
+
+/**
+ * React element to display a modal to merge multiple samples into a single one.
+ * @param {array} samples - list of samples to merge together
+ * @param {boolean} visible - whether the modal is currently visible on the page
+ * @param {function} onComplete - function to call when the merge is complete
+ * @param {function} onCancel - function to call when the merge is cancelled.
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export default function MergeModal({ samples, visible, onComplete, onCancel }) {
+  const { projectId } = useSelector((state) => state.samples);
+  const [merge, { isLoading }] = useMergeMutation();
+
+  const [renameSample, setRenameSample] = React.useState(false);
+  const [error, setError] = React.useState(undefined);
+  const [form] = Form.useForm();
+
+  const initialValues = {
+    primary: samples.valid[0]?.id,
+    newName: "",
+  };
+
+  React.useEffect(() => {
+    if (!renameSample) {
+      form.setFieldsValue({
+        newName: "",
+      });
+    }
+  }, [form, renameSample]);
+
+  // Server validate new name
+  const validateName = async (name) => {
+    if (renameSample) {
+      return serverValidateSampleName(name);
+    } else {
+      return Promise.resolve();
+    }
+  };
+
+  const onSubmit = async () => {
+    let values;
+
+    try {
+      values = await form.validateFields();
+    } catch {
+      return;
+    }
+    const ids = samples.valid
+      .map((sample) => sample.id)
+      .filter((id) => id !== values.primary);
+
+    const { message } = await merge({
+      projectId,
+      request: {
+        ...values,
+        ids,
+      },
+    }).unwrap();
+
+    notification.success({
+      message: i18n("MergeModal.success"),
+      description: message,
+    });
+    onComplete();
+  };
+
+  // TODO: Handle rendering many samples?
+  // TODO: Handle rendering many locked samples?k
```

Comment:
Remove k at the end of the line

Reply:
Removed the comments altogether.  Too many samples would be ridiculous to start with (although I guess if we get an issue we can update it later, but who would merge tons of samples together). Locked samples have been handled on line 177.

Reply:
Yeah true enough. Thanks